src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,505 @@
+            sb.Append('.');
+            sb.Append(_minor);
+            sb.Append('.');
+            sb.Append(_patch);
+            if (_preRelease != null)
+            {
+                sb.Append('-');
+                sb.Append(_preRelease);
+            }
+            if (_commitsAfterRelease != null)
+            {
+                sb.Append('-');
+                sb.Append(_commitsAfterRelease);
+            }
+            if (_commitHash != null)
+            {
+                sb.Append("-g");
+                sb.Append(_commitHash);
+            }
+            return sb.ToString();
+        }
+
+        // static methods
+        /// <summary>
+        /// Parses a string representation of a server version.
+        /// </summary>
+        /// <param name="value">The string value to parse.</param>
+        /// <returns>A server version.</returns>
+        /// <exception cref="FormatException">value</exception>
+        public static ServerVersion Parse(string value)
+        {
+            ServerVersion result;
+            if (TryParse(value, out result))
+            {
+                return result;
+            }
+
+            throw new FormatException($"Invalid ServerVersion: '{value}'.");
+        }
+
+        /// <summary>
+        /// Tries to parse a string representation of a server version.
+        /// </summary>
+        /// <param name="value">The string value to parse.</param>
+        /// <param name="result">The result.</param>
+        /// <returns>True if the string representation was parsed successfully; otherwise false.</returns>
+        public static bool TryParse(string value, out ServerVersion result)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                var pattern = @"(?<major>\d+)\.(?<minor>\d+)(\.(?<patch>\d+)(-(?<preRelease>.*))?)?";
+                var match = Regex.Match((string)value, pattern);
+                if (match.Success)
+                {
+                    var major = int.Parse(match.Groups["major"].Value);
+                    var minor = int.Parse(match.Groups["minor"].Value);
+                    var patch = match.Groups["patch"].Success ? int.Parse(match.Groups["patch"].Value) : 0;
+                    var preRelease = match.Groups["preRelease"].Success ? match.Groups["preRelease"].Value : null;
+
+                    result = new ServerVersion(major, minor, patch, preRelease);
+                    return true;
+                }
+            }
+
+            result = null;
+            return false;
+        }
+
+        // public operators
+        /// <summary>
+        /// Casts a SemanticVersion to a ServerVersion.
+        /// </summary>
+        /// <param name="semanticVersion">The SemanticVersion.</param>
+        /// <returns>A ServerVersion.</returns>
+        /// <exception cref="FormatException">semanticVersion</exception>
+        public static explicit operator ServerVersion(SemanticVersion semanticVersion)
```

Comment:
Remove this explicit conversion operator. It's not used and it would be better not to have it.

Reply:
Done.